### PR TITLE
feat: eliminate performance sensitive

### LINKED ISSUE #796: Eliminate the concept of performance-sensitive mode for users
The performance-sensitive mode introduces additional learning cost for users. We may consider to hide this concept for users. Intuitively, if the index is not built, it is acceptable that the performance slows down. The users only need to be aware of the index building, but not another concept of performance-sensitive mode.

### PR DESCRIPTION
**Pull request by jingchen2222**
* **Please check if the PR fulfills these requirements**

- The commit message follows our guidelines
- Tests for the changes have been added (for bug fixes / features)
- Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
feature

* **What is the current behaviour?** (You can also link to an open issue here)
resolve #796 
  - 1. eliminate set performance sensitive = true|false in cli
  - 2. remove related API in C++ SDK and java SDK
  - 3. add enable_request_performance_sensitive in engine option, for ddl_gen to get a request physical plan performance-insensitive
  - 4. fix yaml case, online batch queries can be executed performance insensitive

* **What is the new behaviour (if this is a feature change)?**

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:

### CODE REVIEW

File: hybridse/examples/toydb/src/cmd/toydb_run_engine.cc [hunk 1]
```diff
@@ -27,9 +27,6 @@ DEFINE_bool(
 DEFINE_bool(enable_expr_opt, true,
             "Specify whether do expression optimization");
 DEFINE_bool(enable_perf_sensitive, true, "enable performance sensitive mode");
```
Comment (aceforeverd) [inline hybridse/examples/toydb/src/cmd/toydb_run_engine.cc]: This opt can rm too

**Comment by zhanghaohit** [comment]
remember to update the related docs

**Comment by jingchen2222** [comment]
> remember to update the related docs

DONE

**Comment by jingchen2222** [comment]
[Reorganize OpenMLDB modes 0.4.0 related work](https://trello.com/c/kISPjYBa/31-reorganize-openmldb-modes-040-related-work)

File: cases/function/window/test_maxsize.yaml [hunk 2]
```diff
@@ -761,8 +753,8 @@ cases:
   -
     id: 26
     desc: 两个union,不同的maxsize
-    mode: rtidb-batch-unsupport,cluster-unsupport
-    tags: ["cluster-执行失败"]
+    mode: cluster-unsupport
+    tags: ["TODO", "@chenjing support online batch", "@chendihao support offline batch", "cluster-执行失败"]
```
Comment (jingchen2222) [inline cases/function/window/test_maxsize.yaml:757]: add issue

**Review by superzhaozheng — APPROVED** [review]
LGTM

File: cases/function/window/test_maxsize.yaml (hunk 2 above)
Comment (jingchen2222) [reply cases/function/window/test_maxsize.yaml:757]: https://github.com/4paradigm/OpenMLDB/issues/869

**Review by aceforeverd — APPROVED** [review]

**Review by dl239 — APPROVED** [review]

**Review by tobegit3hub — APPROVED** [review]
LGTM

**Pull request merged**